File: Mesh4n/trunk/Main/Source/ISyncRepository.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Mesh4n
{
	public interface ISyncRepository
	{
		Sync Get(string id);
		void Save(Sync sync);

		IEnumerable<Sync> GetAll();
		IEnumerable<Sync> GetConflicts();
	}
}

File: Mesh4j/trunk/src/org/mesh4j/sync/adapters/split/ISyncRepository.java
package org.mesh4j.sync.adapters.split;

import java.util.Date;
import java.util.List;

import org.mesh4j.sync.adapters.SyncInfo;
import org.mesh4j.sync.model.IContent;


public interface ISyncRepository {

	void save(SyncInfo syncInfo);

	SyncInfo get(String syncId);

	List<SyncInfo> getAll(Date since, String entityName);

	String newSyncID(IContent content);
}